### PR TITLE
Estevam.insert request

### PR DESCRIPTION
**Pull request by Es7evam**
Add request insertion function.
It is located at dal/requestDAL.go and receives a request, so it can be inserted into the database.

### CODE REVIEW

File: backend/dal/requestDAL.go [hunk 1]
```diff
@@ -19,6 +21,16 @@ func NewRequestDAL(db *sqlx.DB) *RequestDAL {
 	}
 }
 
+// Gets the database connection to be used in requests
+func getDB() *sqlx.DB {
```
Comment (mbotarro) [inline backend/dal/requestDAL.go]: You don't need to get the db in the DAL. A db connection is created once in the main and passed to the DAL through the controller. That's why in the test there's a getDB function that is passed to the DAL when it's created.

File: backend/dal/requestDAL.go [hunk 2]
```diff
@@ -33,3 +45,28 @@ func (dal *RequestDAL) GetLastRequests(before time.Time, size int) ([]models.Req
...
+	fmt.Println(request.MaxPrice)
+	fmt.Println(request.Userid)
+	fmt.Println(request.Categoryid)
+	fmt.Println(request.Timestamp)
+
+	_, err := dal.db.Exec(insertQuery, request.Name, request.Description, request.ExtraInfo, request.MinPrice, request.MaxPrice, request.Userid, request.Categoryid, request.Timestamp)
+
+	// Checks if any error happened during the query execution
+	if err != nil {
+		panic(err)
```
Comment (mbotarro) [inline backend/dal/requestDAL.go]: If there's an error in the db, the whole server will stop. Maybe it'd better to rise the error to the controller or the handler and let the handler treat it. We could send a message to the frontend with the error, for example.

File: backend/dal/requestDAL_test.go [hunk 3]
```diff
@@ -109,3 +109,29 @@ func TestGetLastRequestsBeforeTimestamp(t *testing.T) {
...
+	u := tools.CreateFakeUser(t, db, "user", "user@user.com", "1234", "9999-1111")
+	c := tools.CreateFakeCategory(t, db, "Aula Matemática", "Matemática")
+	req.Userid = u.Userid
+	req.Categoryid = c.ID
+
+	// Executes the test query
+	err := requestDAL.InsertRequest(req)
+
+	// Checks the expected results
+	assert.Equal(t, nil, err)
```
Comment (mbotarro) [inline backend/dal/requestDAL_test.go]: Wouldn't it be better to test if the request were successfully inserted in the db?
You could call the *GetLastRequests* method, for example, and check if the request was inserted correctly.

File: backend/dal/requestDAL.go (hunk 2 above)
Comment (Es7evam) [reply backend/dal/requestDAL.go]: Changed panic into return err, this way the handler will treat the error.

File: backend/dal/requestDAL.go (hunk 1 above)
Comment (Es7evam) [reply backend/dal/requestDAL.go]: Removed function, it shouldn't be there. It wasn't even being used.

File: backend/dal/requestDAL_test.go (hunk 3 above)
Comment (Es7evam) [reply backend/dal/requestDAL_test.go]: Check made with another function comparing the result and the to be inserted request.

File: backend/dal/requestDAL.go [hunk 4]
```diff
@@ -33,3 +34,28 @@ func (dal *RequestDAL) GetLastRequests(before time.Time, size int) ([]models.Req
...
 	return reqs, nil
 }
+
+// InsertRequest Receives a request as a parameter and inserts into the database
+func (dal *RequestDAL) InsertRequest(request models.Request) (models.Request, error) {
+	insertQuery := `INSERT INTO request (name, description, extrainfo, minprice, maxprice, userid, categoryid, timestamp) 
+						VALUES ($1, $2, $3, $4, $5, $6, $7, $8)`
+
+	// Gets the controller of the database and executes the query
+	fmt.Println(request.Name)
```
Comment (mbotarro) [inline backend/dal/requestDAL.go]: We don't need to print the inserted request, do we?

File: backend/dal/requestDAL.go [hunk 5]
```diff
@@ -33,3 +34,28 @@ func (dal *RequestDAL) GetLastRequests(before time.Time, size int) ([]models.Req
...
+	fmt.Println(request.MaxPrice)
+	fmt.Println(request.Userid)
+	fmt.Println(request.Categoryid)
+	fmt.Println(request.Timestamp)
+
+	_, err := dal.db.Exec(insertQuery, request.Name, request.Description, request.ExtraInfo, request.MinPrice, request.MaxPrice, request.Userid, request.Categoryid, request.Timestamp)
+
+	// Checks if any error happened during the query execution
+	if err != nil {
+		return request, err
```
Comment (mbotarro) [inline backend/dal/requestDAL.go]: Maybe it'd be better to return just an empty request in this case models.Request{}

**Pull request merged**